screen/loans/src/main/java/com/ivy/loans/loan/LoanViewModel.kt

```diff
@@ -300,18 +301,28 @@ class LoanViewModel @Inject constructor(
         } ?: defaultCurrencyCode
     }
 
-    private suspend fun calculateAmountPaid(loan: Loan): Double {
+    /**
+     *  Calculates the total amount paid and the total loan amount including any changes made to the loan.
+     *  @return A Pair containing the total amount paid and the total loan amount.
+     */
+    private suspend fun calculateAmountPaidAndTotalAmount(loan: Loan): Pair<Double, Double> {
         val loanRecords = ioThread { loanRecordDao.findAllByLoanId(loanId = loan.id) }
-        var amount = 0.0
+        var amountPaid = 0.0
+        var loanTotalAmount = loan.amount
 
         loanRecords.forEach { loanRecord ->
-            if (!loanRecord.interest) {
-                val convertedAmount = loanRecord.convertedAmount ?: loanRecord.amount
-                amount += convertedAmount
+            if (loanRecord.interest) return@forEach
+            val convertedAmount = loanRecord.convertedAmount ?: loanRecord.amount
+            when (loanRecord.loanRecordType) {
```

Comment:
I've seen this `when(loanRecord.loanRecordType` for a 3rd time, can we create a private extension function inside the VM that does that?

Reply:
I've created the extension function in LoanRecordType.kt file, it's a public pure function. This function is now used in LoanViewModel and LoanDetailsScreen. It makes things a bit neater.